app/routes/api/v1/warehouse/orders/dates/within-delivered/index.ts

```diff
@@ -0,0 +1,39 @@
+import { ActionFunction } from "@remix-run/node";
+import { z } from "zod";
+import {
+  methodNotAllowed,
+  notFoundRequest,
+} from "~/helpers/app-helpers.server";
+import { parseBody } from "~/lib/parse-body.server";
+import { findOrdersWithinDeliveredDates } from "~/models/order/order.server";
+
+export const action: ActionFunction = async ({ request }) => {
+  if (request.method.toLowerCase() != "post") {
+    throw methodNotAllowed();
+  }
+
+  const schema = z.object({
+    startDate: z.nullable(z.date()),
+    endDate: z.nullable(z.date()),
+  });
+
+  const parsedData = await parseBody(request, schema);
+
+  if (!parsedData.startDate) {
+    parsedData.startDate = new Date("1970-01-01");
```

Comment:
Why this?

Reply:
So that you actually provide a date even if in the request it was `null`. (e.g.: if you want to get the orders from the beginning until a specified date)

Reply:
Mumble mumble, make sense but I don't think this is the correct way to do this.

However, i think we should discuss 'bout this in more detail.